spec.md

```diff
@@ -2369,6 +2369,85 @@ As an example `firetruck$y$z`{.firrtl} shares a prefix with
 `firetruck$y`{.firrtl} and `firetruck`{.firrtl}, but does not share a prefix
 with `fire`{.firrtl}.
 
+# Annotations
+
+Annotations encode arbitrary metadata and associate it with zero or more
+targets in a FIRRTL circuit.
```

Comment:
Targets are not yet defined, so I'm not sure if it makes sense to include this without the Targets info

Reply:
oh you did define them. Should this be a Keyword, or should Targets come first?

Reply:
Do Targets make sense in a non-Annotation context? Like do other things besides Annotations make use of Targets?

If not, I think it makes sense to keep targets as a subsection here. I can add a link here to the subsection.